openmdao/core/driver.py

```diff
@@ -1053,10 +1053,18 @@ def record_iteration(requester, prob, case_name):
     outs = model._retrieve_data_of_kind(filt, 'output', 'nonlinear', parallel)
     ins = model._retrieve_data_of_kind(filt, 'input', 'nonlinear', parallel)
 
+    from openmdao.core.problem import Problem
+    if not isinstance(requester, (Problem, Driver)):
+        raise ValueError('Unexpected object requesting recording: {0}'.format(requester))
```

Comment:
Does this happen anywhere?  What could the user do to trigger this?

Reply:
Not that I know of. Just being extra cautious.

Reply:
Since it's not public API, we probably don't need to do this.  The line has no test coverage anyway.

Reply:
ok